File: python/day10.py
import re
import sys

def seeandsay(text):
	matches = re.findall(r'((.)\2*)', text)
	new = ''
	for match in matches:
		new += str(len(match[0])) + match[1]
	return new

text = sys.argv[1]
for i in range(40):
	text = seeandsay(text)
print 'Part 1:' , len(text)

for i in range(10):
	text = seeandsay(text)
print 'Part 2:' , len(text)

File: php/day10.php
<?php

function seeandsay($string)
{
	preg_match_all('/(.)\1*/', $string, $matches);
	$string = '';
	foreach ($matches[0] as $match)
	{
		$string .= strlen($match) . substr($match, 0, 1);
	}
	return $string;
}

$text = '1113122113';
for ($i = 0; $i < 40; $i++)
{
	$text = seeandsay($text);
}

echo 'Part 1: ' . strlen($text) . PHP_EOL;

for ($i = 0; $i < 10; $i++)
{
	$text = seeandsay($text);
}

echo 'Part 2: ' . strlen($text) . PHP_EOL;
